### DISCUSSION TITLE
v4.2.6

### DISCUSSION THREAD

**Discussion by autoSteve** [BJReplay/ha-solcast-solar#219] [Announcements]
Bug fixes, and introducing a Solcast API simulator for collaborators.

## What's Changed
* Fix an issue that was preventing new installs by @autoSteve
* Fix an issue calculating auto-update interval for multi-API key by @autoSteve
* Fix an issue migrating from/to multi-API for Docker setup by @autoSteve
* Fix an issue clearing all forecast history by @autoSteve
* Fix an issue where API count was not incremented on stale start fetch by @autoSteve
* Fix an issue where API used/total & last updated sensors were not updated by @autoSteve
* Add Solcast API simulator to support development and accelerate testing by @autoSteve

Full Changelog: https://github.com/BJReplay/ha-solcast-solar/compare/v4.2.5...v4.2.6

**Comment by zhouska** [comment]
installed 4.2.6 this morning, everything came back after reboot as expected, so far no issues...

**Comment by autoSteve** [comment]
Known issues with v4.2.6:

* If the API key contains a hyphen then it will be considered an invalid key.
* Removing the integration while an auto-update is being performed (which is retrying) will not result in a clean removal. Tasks will remain running until they time out.
* The check for a conflicting integration running is flawed, needs to be overhauled and moved from `__init__.py` to the config flow.

These are already fixed and will be released in v4.2.next.

**Comment by autoSteve** [comment]
v4.2.7 is out.
